torch/_dynamo/variables/tensor.py

```diff
@@ -158,13 +158,16 @@ def specialize(value: torch.Tensor):
                 [int(s) if is_symbolic(s) else s for s in value.size()]
             )
             props["stride"] = tuple(value.stride())
-            props["is_contiguous"] = tuple(
-                [
-                    x
-                    for x in torch._prims_common._memory_formats
-                    if value.is_contiguous(memory_format=x)
-                ]
-            )
+            if torch._C._functorch.is_batchedtensor(value):
+                props["is_contiguous"] = ()
```

Comment:
This should be props["is_contiguous"] = None, right? None causes Dynamo to actually query the value if it shows up. Otherwise it's as if this Tensor isn't contiguous for anything

Reply:
Actually, I'm kind of confused. The thing we want to happen is self.is_contiguous (for this TensorVariable) should be None, so that this branch gets triggered: https://github.com/pytorch/pytorch/blob/49d826bcd3de952eb84a33c89ed399a1a2821c15/torch/_dynamo/variables/tensor.py#L433-L448. What do we need to do to get there?

Reply:
I guess this branch is only executed if one explicitly calls `tensor.is_contiguous()`:

```python
import torch

x = torch.ones(2, 3)

@torch.compile(backend="aot_eager", fullgraph=True)
def wrapper_fn(x):
    if x.is_contiguous():
        return x
    return x + 1

expected = wrapper_fn(x)
```

Reply:
Right, I guess my point is: under Dynamo, if x is a BatchedTensor, then x.is_contiguous(memory_format=channels_last) should raise an error. Same for all other memory_format that are not channels last. Can we add a test for this? It would probably look something like the following:
```
@torch.compile(backend="eager", fullgraph=True)
def f(x):
   if x.is_contiguous(memory_format=channels_last):
       return x.sin()
   return x.cos()

self.assertRaisesRegex(RuntimeError, "something"):
    vmap(f)(x)
```